### DISCUSSION TITLE
Availability of DriftIsolate api on the web

### DISCUSSION THREAD

**Discussion by CrypticVader** [simolus3/drift#2676] [Q&A] [answered]
Hi, I am planning to use Drift for data persistence on an app I'm working on. The docs mention the availability of the DriftIsolate API, which allows accessing the same instance through multiple isolates. I was wondering whether this functionality is supported on the web too.

**Comment by simolus3** [comment, marked as answer]
> I was wondering whether this functionality is supported on the web too.

Isolates aren't supported on the web at all, so drift isolates can't be a thing on the web either. You can use a drift database on web workers though, and the [recommended web setup](https://drift.simonbinder.eu/web/) also uses web workers to offload the database access onto a background worker.

**Reply by CrypticVader** [reply]
That makes complete sense. Thank you for responding so quickly, & also for working on this library.
On an unrelated note, do you think dart to Wasm compilation will bring the isolate API to web?

**Reply by simolus3** [reply]
>  On an unrelated note, do you think dart to Wasm compilation will bring the isolate API to web?

I don't know, I'm not familiar with the state of that implementation. `dart:isolate` is importable, but throws at every method. But since you need to spawn web workers either way (wasm has no builtin concept of threads), you may as well use web APIs like channels directly.
